### ISSUE TITLE
[Bug]: TypeError crash

### ISSUE THREAD

**Issue by Firecul** [ThioJoe/YT-Spammer-Purge#638]
### Duplicate Issues

- [X] There are **no existing posts relating to my problem**
- [ ] There are **existing posts relating to my problem, but the solution given, doesn't work for me.**

### What happened?

I was testing the most recent version to see if it caught the spam threads mentioned here #520 for investments scams with no luck. So I chose to scan names and comment text for the words `bilbery, strategy, invest` just to find those comments. After matching a number of comments the program crashes before writing the log (no more than the title of the log anyway)

### Release version

v2.15.2

### Steps to reproduce

1. Scan https://youtu.be/ao1nONgJ_sY
2. Choose option 6 `Scan both usernames and comment text for criteria you choose`
3. Choose option 2 `An entire string, or multiple strings`
4. Enter `bilbery, strategy, invest` as your search strings

I don't believe the video and specific strings are the cause, I have had this error before when trying to find comments in the #582 example.

### What platform are you seeing this problem on?

Windows (.py script)

### Relevant log output

```shell
Traceback (most recent call last):
  File "C:\Users\firec\Documents\GitHub\YT-Spammer-Purge\YTSpammerPurge.py", line 2258, in <module>
    main()
  File "C:\Users\firec\Documents\GitHub\YT-Spammer-Purge\YTSpammerPurge.py", line 2236, in main
    continueRunning = primaryInstance(miscData)
  File "C:\Users\firec\Documents\GitHub\YT-Spammer-Purge\YTSpammerPurge.py", line 1668, in primaryInstance
    current, logMode, jsonSettingsDict = logging.prepare_logFile_settings(
  File "C:\Users\firec\Documents\GitHub\YT-Spammer-Purge\Scripts\logging.py", line 947, in prepare_logFile_settings
    write_log_heading(current, logMode, filtersDict)
  File "C:\Users\firec\Documents\GitHub\YT-Spammer-Purge\Scripts\logging.py", line 1023, in write_log_heading
    + ", ".join(filterSettings[1]),
TypeError: can only join an iterable
------------------------------------------------
Unknown Error - Code: X-5 occurred. This is probably my fault,
please a bug report on the GitHub issues page, and include the above error info.
Short Link: TJoe.io/bug-report
```

### Screenshots

![WindowsTerminal_Windows_PowerShell (24)](https://user-images.githubusercontent.com/34950371/153735030-d230899f-277b-4549-a6a9-b02d6225d8fc.png)

![notepad++_CUsersfirecDocumentsGitHubYT-Spammer-Purgel](https://user-images.githubusercontent.com/34950371/153735033-a9cbcb47-6d8d-40bc-b1ba-f315fabb88d2.png)

**Comment by ThioJoe** [comment]
Huh, not sure how that one has gone so long without being noticed.

Should be fixed with https://github.com/ThioJoe/YT-Spammer-Purge/commit/898fc09565c91aa58aba95941e7028f08a1487ba for next release

**Comment by ThioJoe** [comment]
Ok should be fixed with latest 2.15.3 release
